src/main/java/seedu/address/logic/parser/tag/AddTagCommandParser.java

```diff
@@ -37,15 +37,17 @@ public AddTagCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_CONTACT, PREFIX_TASK, PREFIX_TAG);
 
-        Index index;
+        Index contactIndex;
+        Index taskIndex;
 
         boolean addTagToContact = argMultimap.getValue(PREFIX_CONTACT).isPresent();
         boolean addTagToTask = argMultimap.getValue(PREFIX_TASK).isPresent();
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            contactIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_CONTACT).orElse("1"));
+            taskIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_TASK).orElse("1"));
```

Comment:
does this cause a bug when user input is index 1?

Reply:
The 1 is essentially a dummy value because there is a boolean addTagToContact and addTagToTask to check if there is a value present in the first place. If there is no value, the index stored is not used.